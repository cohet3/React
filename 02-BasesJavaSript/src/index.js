

const activo = true;

// let mensaje = '';

// if ( !activo ) {
//     mensaje = 'Activo';
// } else {
//     mensaje = 'Inactivo';
// }
// const mensaje = ( activo ) ? 'Activo' : 'Inactivo'; 
// const mensaje = ( activo ) ? 'Activo' : null; 
const mensaje = activo && 'Activo';


console.log(mensaje);

const getUsuarioActivo = ( nombre ) =>({
    uid: 'ABC567',
    username: nombre
})

const usuarioActivo = getUsuarioActivo('Fernando');
console.log( usuarioActivo );
