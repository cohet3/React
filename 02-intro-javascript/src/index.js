console.log('Hola WOrlD!!!');

const persona= {
    nombre:'DAnny',
    apellido : 'Rods',
    edad: 30,
    direccion: {
        ciudad: ' New York',
        Postal:28030
    }

};


console.log(persona)



