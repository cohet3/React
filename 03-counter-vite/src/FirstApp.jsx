import PropTypes from "prop-types";

// const newMessage2 = {
//     nombre:'Daniel',
//     ciudad:'Quito',
//     edad:'20',
// };

export const FirstApp = ({ title, subTitle, name }) => {
  // console.log(props);
  // if ( !title ) {
  //     throw new Error('El title no existe')
  // }

  return (
    <>
      <h2>{title}</h2>
      {/* <code>{ JSON.stringify( newMessage2 ) }</code> */}
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};
FirstApp.defaultProps = {
  title: "no hay titulo",
  subTitle: "no hay subtitulo",
  name: "DAnny Rodas",
};
