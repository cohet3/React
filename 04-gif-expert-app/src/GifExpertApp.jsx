import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";


export const GifExpertApp = () => {
  const [ categories, setCategories ]= useState(['Dragon Ball']);
  
  const onAddCategory = ( newCategory ) => { 
    if (categories.includes(newCategory)) return;
    // console.log(newCategory);  
    // categories.push(newCategory);
      setCategories([ newCategory, ...categories ])

  }
  


  return (
      <>

        <h1>GitExpertApp</h1>
        <AddCategory 
            onNewCategory={ (value) =>onAddCategory(value)}
        />
        
        {
          categories.map( ( category ) => (
            <GifGrid 
            key={ category } 
            category={ category } />
                    
          ))
            
        }
      </>
  )
}

export default GifExpertApp
// WiU973VA3HNP8z6ZbV5UU1carU0cGHFC