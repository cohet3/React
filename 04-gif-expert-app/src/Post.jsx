import React from "react";
import { useParams } from "react-router-dom";
import articlesList from "./components/articleList";

const Articles = () => {
  const { article } = useParams();

  return (
    <>
      <meta charSet="utf-8" />
      <title>Bonzzay - </title>
      <meta name="description" content="fecha" />
      <meta name="keywords" content="post, article" />
      <link rel="canonical" />
      <div className="article-layout_header">
        <div className="article-header__title">
          <b></b>
          <h1></h1>
        </div>
      </div>
      <div className="article-wrapper">
        <p dangerouslySetInnerHTML={{ __html: articlesList[0].description }} />
      </div>
    </>
  );
};

export default Articles;
