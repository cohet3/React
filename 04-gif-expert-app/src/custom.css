body {
  padding: 20px;
}
button {
  margin-right: 15px;
}

a {
  margin-right: 5px;
}
.container-fluid {
  width: 100%;
  max-width: 1200px; /* Puedes ajustar este valor según tus necesidades */
  margin: 0 auto;
  padding: 20px;

  @media (min-width: 992px) {
    padding: 20px;
    width: 100%;
    display: block !important;
  }

  @media (max-width: 991px) and (min-width: 768px) {
    padding: 20px;
    width: 100%;
    display: block !important;
  }
  @media (max-width: 767px) {
    padding: 10px;
    display: block !important;
  }
}

.españa {
  --uv-styles-color-outline: #3c4043;
  --uv-styles-color-scrim: #0009;
  --uv-styles-color-on-scrim: #fff;
  --uv-styles-color-icon-on-scrim: #fff;
  --uv-styles-color-primary: #8ab4f8;
  --uv-styles-color-on-primary: #202124;
  --uv-styles-color-secondary: #394457;
  --uv-styles-color-icon-on-secondary: #9aa0a6;
  --uv-styles-color-on-secondary: #d2e3fc;
  --uv-styles-color-tertiary: #303134;
  --uv-styles-color-on-tertiary: #bdc1c6;
  --uv-styles-color-icon-on-tertiary: #bdc1c6;
  --uv-styles-color-text-primary: #8ab4f8;
  --uv-styles-color-text-emphasis: #e8eaed;
  --uv-styles-color-icon-emphasis: #e8eaed;
  --uv-styles-color-text-de-emphasis: #9aa0a6;
  --uv-styles-color-text-default: #bdc1c6;
  --uv-styles-color-icon-default: #bdc1c6;
  --uv-styles-color-review-stars: #fdd663;
  --uv-styles-color-review-stars-empty: #80868b;
  --uv-styles-color-letterbox: #000;
  --uv-styles-color-surface: #171717;
  --uv-styles-color-footer: #171717;
  --lens-transparent-image-checkerboard-dark: rgb(232, 234, 237);
  --lens-transparent-image-checkerboard-light: #fff;
  --lens-title-text-color: var(--lens-text-color-primary);
  --lens-subtitle-text-color: var(--lens-text-color-secondary);
  --lens-label-text-color: var(--lens-text-color-secondary);
  --lens-description-text-color: var(--lens-text-color-secondary);
  --lens-feedback-text-color: var(--lens-text-color-secondary);
  --safesearch-banner-border-color: rgb(241, 243, 244);
  --lens-copy-button-text-color: rgb(60, 64, 67);
  --lens-copy-button-icon-color: rgb(26, 115, 232);
  --lens-copy-button-container-fill-color: #fff;
  --lens-language-list-divider-color: rgb(232, 234, 237);
  --lens-llm-accent2-color: rgb(150, 90, 255);
  --lens-title1-font-size: calc(18 / var(--lens-base-font-size, 16) * 1rem);
  --lens-title2-font-size: calc(16 / var(--lens-base-font-size, 16) * 1rem);
  --lens-title3-font-size: calc(14 / var(--lens-base-font-size, 16) * 1rem);
  --lens-subtitle1-font-size: calc(14 / var(--lens-base-font-size, 16) * 1rem);
  --lens-label-font-size: calc(12 / var(--lens-base-font-size, 16) * 1rem);
  --lens-body1-font-size: calc(14 / var(--lens-base-font-size, 16) * 1rem);
  --lens-body2-font-size: calc(12 / var(--lens-base-font-size, 16) * 1rem);
  --lens-popular-brand-color: rgb(252, 173, 112);
  --lens-background-color-primary: rgb(32, 33, 36);
  --lens-background-color-secondary: rgb(95, 99, 104);
  --lens-background-image-fill-color: rgb(60, 64, 67);
  --lens-background-image-fill-with-scrim-color: rgb(54, 58, 61);
  --lens-image-scrim-color: rgba(255, 255, 255, 0.03);
  --loading-skeleton-color: rgb(60, 64, 67);
  --loading-shine-color: rgb(95, 99, 104);
  --luxe-secondary-color: rgba(57, 68, 87, 1);
  --lens-text-color-primary: rgb(232, 234, 237);
  --lens-text-color-secondary: rgb(154, 160, 166);
  --lens-text-color-tertiary: rgb(232, 234, 237);
  --lens-text-color-disabled: rgb(128, 134, 139);
  --lens-link-color-primary: rgb(138, 180, 248);
  --lens-link-color-secondary: rgb(210, 227, 252);
  --lens-link-color-tertiary: rgb(138, 180, 248);
  --lens-failure-info-text-color: rgb(232, 234, 237);
  --lens-divider-color-primary: rgb(60, 64, 67);
  --lens-bold-divider-color: rgb(23, 23, 23);
  --lens-border-color-primary: rgb(95, 99, 104);
  --lens-icon-color-primary: rgb(232, 234, 237);
  --lens-gtin-icon-background-color: rgb(32, 33, 36);
  --lens-gtin-icon-stroke-color: #dadada;
  --lens-search-button-outline-color-lens: rgb(23, 78, 166);
  --lens-search-button-container-fill-color-lens: var(--luxe-secondary-color);
  --lens-search-button-outline-color-image-search: rgb(95, 99, 104);
  --lens-search-button-container-fill-color-image-search: rgba(0, 0, 0, 0);
  --lens-search-button-border-color-image-search: rgb(95, 99, 104);
  --lens-more-images-button-highlight-color: rgba(232, 234, 237, 0.1);
  --lens-pill-button-container-fill-color: rgb(60, 64, 67);
  --lens-inline-toggle-button-hover-fill-color: rgba(255, 255, 255, 0.04);
  --lens-inline-toggle-icon-color: rgb(232, 234, 237);
  --lens-text-icon-button-color: rgb(232, 234, 237);
  --lens-multimodal-thumbnail-scrim-color: rgba(255, 255, 255, 0.04);
  --lens-multimodal-search-container-fill-color: rgb(60, 64, 67);
  --lens-multimodal-mobile-search-container-fill-color: rgba(60, 64, 67);
  --lens-multimodal-search-icon-color: rgb(102, 157, 246);
  --lens-multimodal-clear-icon-color: rgb(232, 234, 237);
  --lens-multimodal-divider-color: rgb(95, 99, 104);
  --lens-multimodal-tooltip-color: rgb(174, 203, 250);
  --lens-multimodal-back-button-color: rgba(168, 199, 250);
  --lens-multimodal-placeholder-text-color: rgba(150, 155, 161);
  --lens-multimodal-mobile-box-shadow-color: rgba(23, 23, 23, 0.32);
  --lens-multimodal-uv-shadow-top: rgba(0, 0, 0, 0.15);
  --lens-multimodal-uv-shadow-bottom: rgba(0, 0, 0, 0.3);
  --lens-fill-button-ripple-color: #fff;
  --lens-text-button-active-text-color: rgb(174, 203, 250);
  --lens-text-button-ripple-color: rgb(174, 203, 250);
  --lens-answer-card-ripple-press-opacity: 0.1;
  --lens-ris-item-text-color: rgb(232, 234, 237);
  --safesearch-banner-title-text-color: rgb(232, 234, 237);
  --lens-chip-fill-color-selected: var(--luxe-secondary-color);
  --lens-chip-ripple-color: rgb(232, 234, 237);
  --lens-chip-text-color-assistive: rgb(232, 234, 237);
  --lens-chip-text-color-suggestive: rgb(154, 160, 166);
  --lens-chip-text-color-selected: rgb(210, 227, 252);
  --lens-result-panel-loading-gif-url: url(https://www.gstatic.com/lens/web/ui/slurploading_darkmode-0f3c762403f6ab2c4528aedf06f37849.gif);
  --lens-scroll-fab-icon-color: var(--lens-icon-color-primary);
  --lens-scroll-fab-background-color: rgb(60, 64, 67);
  --lens-scroll-fab-ink-color: #fff;
  --lens-language-picker-background-color: rgb(32, 33, 36);
  --lens-language-picker-text-color: #fff;
  --lens-language-picker-chip-background-color: var(--luxe-secondary-color);
  --lens-language-picker-arrow-color: rgb(232, 234, 237);
  --lens-language-list-subheader-color: rgb(66, 133, 244);
  --lens-language-list-icon-color: #fff;
  --lens-language-list-selected-background-color: var(--luxe-secondary-color);
  --lens-language-list-detected-locale-text-color: rgb(189, 193, 198);
  --lens-background-text-panel-fill-color: rgba(255, 255, 255, 0.08);
  --lens-background-text-panel-text-color: #fff;
  --lens-grid-item-favicon-background-color: #fff;
  --lens-grid-item-highlight-color: rgba(232, 234, 237, 0.1);
  --lens-grid-item-cluster-border-color: rgb(95, 99, 104);
  --lens-grid-item-cluster-highlight-color: rgba(138, 180, 248, 0.1);
  --lens-grid-item-cluster-click-color: rgba(138, 180, 248, 0.2);
  --lens-snackbar-background-color: #fff;
  --lens-snackbar-text-color: rgb(60, 64, 67);
  --lens-llm-card-background-color: rgb(50, 47, 56);
  --lens-llm-card-text-color: rgb(244, 238, 250);
  --lens-llm-header-icon-color: rgb(208, 188, 255);
  --lens-llm-header-text-color: rgb(244, 238, 250);
  --lens-llm-selected-container-color: rgb(76, 62, 118);
  --lens-llm-primary-color: rgb(207, 189, 255);
  --lens-llm-complement-color: rgb(247, 243, 185);
  --lens-on-background-color: rgb(232, 234, 237);
  --lens-background-color: rgb(32, 33, 36);
  --lens-on-surface-color: rgb(232, 234, 237);
  --lens-on-surface-variant-color: rgb(154, 160, 166);
  --lens-surface-color: rgb(32, 33, 36);
  --lens-surface-variant-color: rgb(60, 64, 67);
  --lens-surface-variant-2-color: rgba(37, 38, 41);
  --lens-on-primary-color: rgb(32, 33, 36);
  --lens-primary-color: rgb(138, 180, 248);
  --lens-on-secondary-color: rgb(210, 227, 252);
  --lens-secondary-color: sass-color.mix(
    rgb(32, 33, 36),
    rgb(138, 180, 248),
    24%
  );
  --lens-stars-color: rgb(253, 214, 99);
  --lens-popular-color: rgb(250, 123, 23);
  --lens-on-error-color: rgb(32, 33, 36);
  --lens-outline-color: rgb(95, 99, 104);
  --lens-error-color: rgb(242, 139, 130);
  --lens-informative-color: rgb(128, 134, 139);
  --lens-in-stock-color: rgb(129, 201, 149);
  --lens-out-of-stock-color: rgb(242, 139, 130);
  --lens-official-color: rgb(154, 160, 166);
  --lens-used-color: rgb(154, 160, 166);
  color-scheme: dark;
  font-family: arial, sans-serif;
  font-size: 14px;
  color: #bdc1c6;
  --center-width: 652px;
  --rhs-margin: 76px;
  --rhs-width: 372px;
  --lhs-refinements-width: 0px;
  --lhs-margin: 24px;
  --lhs-width: calc(
    var(--center-abs-margin) - var(--lhs-margin) - var(--center-margin)
  );
  --center-margin: 24px;
  --grid-gutter-width: 20px;
  --grid-column-width: 36px;
  --center-abs-margin: calc(25vw + -188.75px);
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  overflow: hidden;
  position: absolute;
  border-radius: inherit;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
