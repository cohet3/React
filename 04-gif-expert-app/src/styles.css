/* * {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}
:root {
  --height: 350px;
  --widht: 450px;
  --important-color: #e8175d;
  --shadow-color: 232, 23, 93;
  --shadow1: rgba(var(--shadow-color), 1.2) 0px 1px 1px 0px inset;
  --shadow2: rgba(var(--shadow-color), 0.35) 0px 50px 100px -20px;
  --shadow3: rgba(var(--shadow-color), 1.3) 0px 30px 60px -30px;
}

.container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-content: center;
}

.container-card {
  height: 450px;
  width: 350;

  align-content: center;
  border-radius: 7px;
  overflow: hidden;
  box-shadow: rgba(255, 255, 255, 0.1) 0px 1px 1px 0px inset,
    rgba(50, 50, 93, 0.25) 0px 50px 100px -20px,
    rgba(0, 0, 0, 0.3) 0px 30px 60px -30px;
  margin: 1em;
  transition: 200ms box-shadow ease-in;
}

.container-header {
  display: flex;
  justify-content: center;
  height: 300px;
  width: 400px;
}

img.container-card__img {
  object-fit: cover;
  height: 300px;
  width: 460px;
}

.container-card__text {
  text-align: center;
  font-size: 1.2rem;
  margin-top: 2em;
}

.container-card__text::first-letter {
  font-size: 1.5rem;
  color: var(--important-color);
  font-weight: bolder;
}

.category-text {
  text-align: center;
  font-size: 2.5rem;
}

.category-text::first-letter {
  font-size: 3rem;
  color: var(--important-color);
}

.container-card:active {
  box-shadow: var(--shadow1), var(--shadow2), var(--shadow3);
}

@media (hover: hover) {
  .container-card:hover {
    box-shadow: var(--shadow1), var(--shadow2), var(--shadow3);
  }
}
.barra-desplegable {
  border: 5px;
  display: inline-block;
  position: relative;
}
.objetos {
  background-color: aqua;
  border: 5px 5px 5px 5px;
  margin: 10px 10px 10px 10px;
  padding: 8px;
}
*/
/* menu

#menu ul {
  list-style:none;
  margin:0;
  padding:0;
 }
 
 /* items del menu */

/* #menu ul li {
  background-color: #2e518b;
}

/* enlaces del menu */

/* #menu ul a {
  display: block;
  color: #100d0d;
  text-decoration: none;
  font-weight: 400;
  font-size: 15px;
  padding: 10px;
  font-family: "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
  text-transform: uppercase;
  letter-spacing: 1px;
}

/* items del menu */

/*#menu ul li {
  position: relative;
  float: left;
  margin: 0;
  padding: 0;
}

/* efecto al pasar el ratón por los items del menu */

#menu ul li:hover {
  background: #1a61bd;
}
/* menu desplegable */

/*#menu ul ul {
  display: none;
  position: absolute;
  top: 100%;
  left: 0;
  background: #111111;
  padding: 0;
}

/* items del menu desplegable */

/*#menu ul ul li {
  float: none;
  width: 150px;
}

/* enlaces de los items del menu desplegable */

/*#menu ul ul a {
  line-height: 120%;
  padding: 10px 15px;
}

/* items del menu desplegable al pasar el ratón */

/*#menu ul li:hover > ul {
  display: block;
}

/* estilos de clickprinting */
/*.general {
  width: 570px;
  height: 620px;
  border: 1px solid var(--grises-gris-15-borde, #0e0d0d);
  background: var(--grises-gris-3-bg-total, #f8f8f8);
}
.Redonda {
  width: 64px;
  height: 64px;
  flex-shrink: 0;
  border-radius: 64px;
  background: url(<path-to-image>), lightgray 50% / cover no-repeat;
} */
/* //Upload details */
body {
  border: 50px black;
  margin: 50px;
  padding: 5%;
  background-color: #ebf0f1;
  display: flex;

  .detailwork {
    background-color: white;
    border: 5px solid rgb(14, 13, 13);
    margin-top: 12px;
    padding: 16px;

    .headDetailwork {
      display: flex;
      margin-bottom: 10px;
      background-color: rgb(251, 250, 250);

      .imgRound {
        display: flex;
        width: 94px;
        height: auto;
        flex-direction: column;
        flex-shrink: 0;

        > img {
          display: block;
          width: 64px;
          height: 64px;
          border-radius: 50%;
        }
      }
    }
    .infoHeadDetailWork {
      display: flex;
      width: 247px;
      height: auto;
      flex-direction: column;
      flex-shrink: 0;

      > h3 {
        width: 100%;
        font-size: 10px;
        color: skyblue;
        display: flex;
        line-height: 22px;
        text-transform: uppercase;
      }
      > h2 {
        color: rgb(9, 9, 9);
        width: 100%;
        font-size: 16px;
        line-height: 22px;
        display: flex;
        margin-top: -5px;
      }
      > span {
        color: gray;
        width: 100%;
        font-size: 14px;
        line-height: 22px;
        display: flex;
        margin-top: 5px;

        > b {
          color: gray;
          line-height: 23px;
          font-weight: 300;
          margin-left: 10px;
        }
      }
    }
    .totalcount {
      display: flex;
      background-color: gray;
      height: fit-content;
      border-radius: 4px;
      flex-direction: row-reverse;
      .inner_form_blk {
        background-color: blue;
        .inner_num_bg {
          background-color: forestgreen;
          .inner_text {
            width: auto;
            height: auto;
            border-radius: 4px;
            background: gray;
            display: flex;
            flex-direction: column-reverse;
            .h6 {
              color: #1a1a1a;
            }
          }
        }
      }
    }
  }

  .instruccionesPlantilla {
    background: #ffffff;
    display: grid;
    width: 100%;
    ul {
      display: -webkit-inline-box;
      li {
        padding: 0;
        margin: 0;
        width: 50%;
        border: solid 1px #d9d9d9;
        transition: all 0.2s linear;
        .on {
          box-sizing: border-box;
          cursor: pointer !important;
          a {
            padding: 16px 0 12px 19px;
            margin: 0;
            clear: both;
            display: inline-block;
            cursor: default;
            width: 100%;
            box-sizing: border-box;
            cursor: pointer !important;
            .bookico {
              transition: all 0.2s linear;
              background-color: #25b1bf !important;
            }
            span {
              color: #999 !important;

              -moz-transition: all 0.2s linear;
              -o-transition: all 0.2s linear;
              transition: all 0.2s linear;
            }
          }
        }
      }
    }
  }
}
.uploadDetails {
  width: auto;
  height: auto;
  display: flex;
  margin: 10px;
  > h1 {
    color: #0e0d0d;
  }
  > input[type="file"] {
    color: #25b1bf;
    height: auto;
    width: auto;
  }
  > input[type="number"]::-webkit-inner-spin-button,
  input[type="number"]::-webkit-outer-spin-button {
    background-color: rgb(244, 244, 244);
    width: auto;
    height: 10%;
    display: flex;
  }

  span:nth-child(1) {
    font-size: 13px;
    font-family: Source Sans Pro;
    font-style: normal;
    font-weight: 400;
    line-height: 15px;
    color: gray;
    display: block;
    padding-top: 8px;
  }
  span:nth-child(2) {
    font-size: 18px;
    font-family: Source Sans Pro;
    font-style: normal;
    font-weight: 400;
    line-height: 24px;
    color: gray;
    display: block;
  }
}
li:last-child {
  border-bottom: 1px solid gray;
}
li.active {
  background-color: white;
  border-left: 4px solid blue;
  border-right: none;
  span:nth-child(1) {
    color: blue;
  }
  span:nth-child(2) {
    color: gray;
  }
}

.tramitar_pedido {
  background-color: gray;
  width: 100%;
  height: 56px;
  border-radius: 4px;
  display: block;
  color: var(--grises-blanco, #fff);
  text-align: center;
  font-family: Source Sans Pro;
  font-size: 18px;
  font-style: normal;
  font-weight: 600;
  line-height: 24px;
  letter-spacing: 0.5px;
  margin-top: 32px;
  padding-bottom: 16px;
}
.impresion {
  width: 100%;
  float: left;

  padding: 20px;
  box-sizing: border-box;
  position: relative;
  background-color: white;
  ul {
    list-style-type: none;
    width: 100%;
    display: flex;
    flex-wrap: wrap;

    li {
      width: 50%;
      float: left;

      color: #1a1a1a;
      font-family: Source Sans Pro;
      font-size: 14px;
      font-style: normal;
      font-weight: 600;
      line-height: 22.4px;
    }

    li:nth-child(odd) {
      border-bottom: 0px;
      font-weight: 700;
      color: gray;

      color: #1a1a1a;
      font-family: Source Sans Pro;
      font-size: 14px;
      font-style: normal;
      font-weight: 400;
      line-height: 22.4px;
    }
  }
}
