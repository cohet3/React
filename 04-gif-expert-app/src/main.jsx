import React from "react";
import ReactDOM from "react-dom/client";
import { GifExpertApp } from "./GifExpertApp";
import "./styles.css";
import Pruebacss from "./components/Pruebacss";
import Camara from "./components/Camara";
import ClickPrinting from "./components/ClickPrinting";

import "./custom.css";
import CounterApp from "./components/Hook-App/01-useState/CounterApp";
import CounterWithCustomHook from "./components/Hook-App/01-useState/CounterWithCustomHook";
import { SimpleForm } from "./components/Hook-App/02-useEffect/SimpleForm";
import { FormWithCustomHook } from "./components/Hook-App/02-useEffect/FormWithCustomHook";
import { MultipleCustomHooks } from "./components/Hook-App/03-examples/MultipleCustomHooks";
import PruebaBou from "./components/PruebaBou";
import { FocusScreem } from "./components/Hook-App/04-useRef/FocusScreem";
import { Layout } from "./components/Hook-App/05-useLayoutEffect/Layout";
import { Memorize } from "./components/Hook-App/06-memos/Memorize";
import { MemoHook } from "./components/Hook-App/06-memos/MemoHook";
import { CallbackHook } from "./components/Hook-App/06-memos/CallbackHook";
import { Padre } from "./components/Hook-App/07-tarea-memo/Padre";
import "./components/Hook-App/08-useReducer/intro-reducer";
import { TodoApp } from "./components/Hook-App/08-useReducer/TodoApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <>
    {/* <GifExpertApp /> */}
    {/* <Pruebacss /> */}
    {/* <PruebaBou /> */}
    {/* <Camara /> */}
    {/* <ClickPrinting /> */}
    {/* <Hook />
     <CounterApp /> */}
    {/* <CounterWithCustomHook /> */}
    {/* <SimpleForm /> */}
    {/* <FormWithCustomHook /> */}
    {/* <MultipleCustomHooks /> */}
    {/* <FocusScreem />  */}
    {/* <Layout /> */}
    {/* <Memorize /> */}
    {/* <MemoHook /> */}
    {/* <CallbackHook /> */}
    {/* <Padre /> */}
    <TodoApp />
  </>
);
