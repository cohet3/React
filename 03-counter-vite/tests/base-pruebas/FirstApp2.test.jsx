import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/FirstApp";

describe('Pruebas en zFirestApp />', () => {

  const title= 'Hola, Soy Goku';
  const subTitle = 'Soy un subtitulo';

  test('Debe de hacer match con el snapshot', () => {

    const { container } = render( <FirstApp title={ title } />);
    expect( container ).toMatchSnapshot();
    
  });
  test('debe de mostar el mensaje "Hola, Soy Goku"', () =>{
  
    render( <FirstApp title={ title } />);
    expect(screen.getByText(title) ).toBeTruthy();
    // screen.debug();
  });
  test('debe mostrar un titulo en un h1', () => {
    render( <FirstApp title={ title } />);
    expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title ) 
    
  });
  test('Debe de mostar el sybtitulo enviado por props', () => {
    render(
      <FirstApp
      title={ title }
      subTitle={ subTitle }
      />
    );
    expect( screen.getAllByText(subTitle).length ).toBe(2)
    
  });
  
  
  
  
})
