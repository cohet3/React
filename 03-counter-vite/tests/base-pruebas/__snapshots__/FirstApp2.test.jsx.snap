// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en zFirestApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    DAnny Rodas
  </p>
</div>
`;
